bundles/org.eclipse.ui.ide.application/src/org/eclipse/ui/internal/ide/application/DelayedEventsProcessor.java

```diff
@@ -183,90 +198,119 @@ public void run() {
 				IWorkbenchWindow window = PlatformUI.getWorkbench().getActiveWorkbenchWindow();
 				if (window == null)
 					return;
-				// System.err.println(System.currentTimeMillis());
 				FileLocationDetails details = FileLocationDetails.resolve(initialPath);
 				if (details == null || !details.fileInfo.exists()) {
 					String msg = NLS.bind(IDEWorkbenchMessages.OpenDelayedFileAction_message_fileNotFound, initialPath);
 					MessageDialog.open(MessageDialog.ERROR, window.getShell(),
 							IDEWorkbenchMessages.OpenDelayedFileAction_title, msg, SWT.SHEET);
-				} else if (details.fileInfo.isDirectory()) {
-					SmartImportWizard wizard = new SmartImportWizard();
-					wizard.setInitialImportSource(new File(details.fileStore.toURI()));
-					WizardDialog dialog = new WizardDialog(window.getShell(), wizard);
-					dialog.setBlockOnOpen(false);
-					dialog.open();
-				} else {
-					IWorkbenchPage page = window.getActivePage();
-					if (page == null) {
-						String msg = NLS.bind(IDEWorkbenchMessages.OpenDelayedFileAction_message_noWindow,
-								details.path);
-						MessageDialog.open(MessageDialog.ERROR, window.getShell(),
-								IDEWorkbenchMessages.OpenDelayedFileAction_title, msg, SWT.SHEET);
-					}
-					try {
-						IEditorPart openEditor = IDE.openInternalEditorOnFileStore(page, details.fileStore);
-						Shell shell = window.getShell();
-						if (shell != null) {
-							if (shell.getMinimized())
-								shell.setMinimized(false);
-							shell.forceActive();
-						}
-
-						if (details.line >= 1) {
-							try {
+					return;
+				}
+				IWorkbenchPage page = window.getActivePage();
+				if (page == null) {
+					String msg = NLS.bind(IDEWorkbenchMessages.OpenDelayedFileAction_message_noWindow,
+							details.path);
+					MessageDialog.open(MessageDialog.ERROR, window.getShell(),
+							IDEWorkbenchMessages.OpenDelayedFileAction_title, msg, SWT.SHEET);
+					return;
+				}
+				Shell shell = window.getShell();
+				if (shell != null) {
+					if (shell.getMinimized())
+						shell.setMinimized(false);
+					shell.forceActive();
+				}
+				try {
+					if (details.fileInfo.isDirectory()) {
+						// Open a directory
+						File directory = new File(details.fileStore.toURI());
+						for (IProject project : ResourcesPlugin.getWorkspace().getRoot().getProjects()) {
```

Comment:
May it be that some of the methods in `ResourcesPlugin.getWorkspace().getRoot().getProjects()` return null?

Reply:
no